tests/plugins_t/test_prometheusmonitoring.py

```diff
@@ -0,0 +1,143 @@
+from tardis.plugins.prometheusmonitoring import PrometheusMonitoring
+from tardis.resources.dronestates import RequestState
+from tardis.utilities.attributedict import AttributeDict
+from tardis.interfaces.siteadapter import ResourceStatus
+
+from datetime import datetime
+from unittest import TestCase
+from unittest.mock import Mock
+from unittest.mock import patch
+
+from ..utilities.utilities import run_async
+
+
+class TestPrometheusMonitoring(TestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls.mock_config_patcher = patch(
+            "tardis.plugins.prometheusmonitoring.Configuration"
+        )
+        cls.mock_config = cls.mock_config_patcher.start()
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.mock_config_patcher.stop()
+
+    @patch("tardis.plugins.prometheusmonitoring.logging", Mock())
+    def setUp(self):
+        self.config = self.mock_config.return_value
+        self.config.Plugins.PrometheusMonitoring.addr = "127.0.0.1"
+        self.config.Plugins.PrometheusMonitoring.port = 1234
+
+        self.plugin = PrometheusMonitoring()
+
+    @patch("tardis.plugins.prometheusmonitoring.logging", Mock())
+    def test_notify(self):
+        test_param = AttributeDict(
+            site_name="test-site",
+            machine_type="test_machine_type",
+            created=datetime.now(),
+            updated=datetime.now(),
+            drone_uuid="12345666",
+            resource_status=ResourceStatus.Booting,
+        )
+        test_state = RequestState()
+
+        run_async(self.plugin.notify, test_state, test_param)
+
+        assert self.plugin._booting.get({}) == 1.0
+        assert self.plugin._booting.get({}) == 1.0
+        assert self.plugin._running.get({}) == 0.0
+        assert self.plugin._stopped.get({}) == 0.0
+        assert self.plugin._deleted.get({}) == 0.0
+        assert self.plugin._error.get({}) == 0.0
+
+        test_param = AttributeDict(
```

Comment:
I would suggest to define a function `get_test_param` in order to avoid code duplication here.

```python
def get_test_param(drone_uuid, resource_status):
    return AttributeDict(
                site_name="test-site",
                machine_type="test_machine_type",
                created=datetime.now(),
                updated=datetime.now(),
                drone_uuid=drone_uuid,
                resource_status=resource_status,
            )
```

Reply:
Fixed in 5a07a6a.